### PR TITLE
[2.8] [HttpFoundation] HttpCache refresh stale responses containing an ETag

### PR DESCRIPTION
**Pull request by maennchen**
| Q | A |
| --- | --- |
| Branch? | 2.8 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #19390 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php [hunk 1]
```diff
@@ -600,6 +600,9 @@ protected function lock(Request $request, Response $entry)
      */
     protected function store(Request $request, Response $response)
     {
+        if (!$response->headers->has('Date')) {
```
Comment (maennchen) [inline src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php]: I wanted to explicitly call `setDate` to clarify what the code is doing even though one line containing the following would have the same effect:

```php
$response->getDate();
```

**Comment by jameshalsall** [comment]
Shouldn't this go into 2.7?

**Comment by maennchen** [comment]
I've never conrtibuted to symfony core before and I'm not sure how the merging will go.
Here'd be a 2.7 Branch: https://github.com/maennchen/symfony/tree/hotfix/2.7_httpcache_ttl_fix

**Comment by maennchen** [comment]
Replaced by #19397

**Pull request closed without merging**